plutus-exe/src/Main.hs

```diff
@@ -50,25 +54,39 @@ plutusOpts = hsubparser (
 typecheckOpts :: Parser TypecheckOptions
 typecheckOpts = TypecheckOptions <$> input
 
+evalMode :: Parser EvalMode
+evalMode = option auto
+  (  long "mode"
+  <> short 'm'
+  <> metavar "MODE"
+  <> value CEK
+  <> showDefault
+  <> help "Evaluation mode (one of CK or CEK)" )
+
 evalOpts :: Parser EvalOptions
-evalOpts = EvalOptions <$> input
+evalOpts = EvalOptions <$> input <*> evalMode
 
 main :: IO ()
 main = do
     options <- customExecParser (prefs showHelpOnEmpty) plutus
     case options of
         Typecheck (TypecheckOptions inp) -> do
             contents <- getInput inp
-            case (PC.runQuoteT . PC.parseTypecheck 1000 . BSL.fromStrict . encodeUtf8 . T.pack) contents of
+            let bsContents = (BSL.fromStrict . encodeUtf8 . T.pack) contents
```

Comment:
This is fine, but generally we should have a `Text`-based API. Maybe in addition to the `ByteString` one, but a `ByteString` is really just a sequence of bytes, it's strange to ask the end user to convert his user-written `Text` into a `ByteString`.

Reply:
I 100% agree, there are a bunch of places we do pointless converting back and forth between bytestrings for this reason.